controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1740,6 +1758,33 @@ func saveHostProvisioningSettings(host *metal3api.BareMetalHost, info *reconcile
 	return dirty, nil
 }
 
+func (r *BareMetalHostReconciler) saveHostFirmwareComponents(prov provisioner.Provisioner, info *reconcileInfo, hfc *metal3api.HostFirmwareComponents) (dirty bool, err error) {
+	dirty = false
+	if !reflect.DeepEqual(hfc.Status.Updates, hfc.Spec.Updates) {
+		info.log.Info("HostFirmwareComponents Status", "updates", hfc.Status.Updates)
+
+		for i := range hfc.Spec.Updates {
+			hfc.Spec.Updates[i].DeepCopyInto(&hfc.Status.Updates[i])
```

Comment:
This looks like it assumes that Spec.Updates and Status.Updates are always the same length.

Reply:
yeah, since what is in the Spec.Updates is what we would like to have in Status.Updates

Reply:
We want it to be the same _after_ we copy.
This is assuming it has the same length _before_.
I think this code will crash.

Reply:
maybe I can just  have `Status.Updates` with the same size doing something like

```
hfc.Status.Updates = make([]metal3api.FirmwareUpdate, len(hfc.Spec.Updates))
for i := range hfc.Spec.Updates {
     hfc.Spec.Updates[i].DeepCopyInto(&hfc.Status.Updates[i])
}
```

?

Reply:
Or even simpler just make an empty list and `append` DeepCopy results into it. But your variant will probably work as well.

Reply:
yes, either of those is fine